DTAConfig/ScreenResolution.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.Xna.Framework.Graphics;
+
+namespace DTAConfig
+{
+    /// <summary>
+    /// A single screen resolution.
+    /// </summary>
+    public sealed record ScreenResolution : IComparable<ScreenResolution>
+    {
+        public ScreenResolution(int width, int height)
+        {
+            Width = width;
+            Height = height;
+        }
+
+        /// <summary>
+        /// The width of the resolution in pixels.
+        /// </summary>
+        public int Width { get; set; }
+
+        /// <summary>
+        /// The height of the resolution in pixels.
+        /// </summary>
+        public int Height { get; set; }
+
+        public override string ToString()
+        {
+            return Width + "x" + Height;
+        }
+
+        public void Deconstruct(out int width, out int height)
+        {
+            width = this.Width;
+            height = this.Height;
+        }
+
+        public static implicit operator ScreenResolution((int Width, int Height) resolutionTuple) => new(resolutionTuple.Width, resolutionTuple.Height);
+
+        public static implicit operator (int Width, int Height)(ScreenResolution resolution) => new(resolution.Width, resolution.Height);
+
+        public static implicit operator ScreenResolution(string resolution)
+        {
+            List<int> resolutionList = resolution.Trim().Split('x').Take(2).Select(int.Parse).ToList();
+            return new(resolutionList[0], resolutionList[1]);
+        }
+
+        public static implicit operator string(ScreenResolution resolution) => resolution.ToString();
+
+        public bool Fit(ScreenResolution child)
+        {
+            return this.Width >= child.Width && this.Height >= child.Height;
+        }
+
+        public int CompareTo(ScreenResolution res2)
+        {
+            if (this.Width < res2.Width)
+                return -1;
+            else if (this.Width > res2.Width)
+                return 1;
+            else // equal
+            {
+                if (this.Height < res2.Height)
+                    return -1;
+                else if (this.Height > res2.Height)
+                    return 1;
+                else return 0;
+            }
+        }
+
+        // Accessing GraphicsAdapter.DefaultAdapter requiring DXMainClient.GameClass has been constructed. Lazy loading prevents possible null reference issues for now.
+        private static ScreenResolution _desktopResolution = null;
+        public static ScreenResolution DesktopResolution => _desktopResolution ??= new(GraphicsAdapter.DefaultAdapter.CurrentDisplayMode.Width, GraphicsAdapter.DefaultAdapter.CurrentDisplayMode.Height);
+
+        // The default graphic profile supports resolution up to 4096x4096. The number gets even smaller in practice. Therefore, we select 3840 as the limit.
+        public static ScreenResolution HiDefLimitResolution { get; } = "3840x3840";
```

Comment:
Since `DesktopResolution` must be implemented as a property due to lazy loading, other *Resolutions are also made properties to ensure consistency.

Also `PropertyName {get;} = getval();` is a syntactic sugar of `val = getval(); PropertyName => val`. It's different from `PropertyName=>getval()` and thus should not make performance differences.

Reply:
I know. I was suggesting `public static readonly Value = Something()` which is effectively the same as you've written, or const instead of readonly when it's suitable.

Reply:
I prefer properties here to make all *Resolutions consistent